avm/res/document-db/mongo-cluster/main.bicep

```diff
@@ -0,0 +1,575 @@
+
+@description('Optional. Whether high availability is enabled on the node group.')
+param highAvailabilityMode bool = false
+
+@description('Required. Number of nodes in the node group.')
+param nodeCount int
+
+@description('Optional. Deployed Node type in the node group.')
+param nodeType string = 'Shard'
+
+@description('Required. SKU defines the CPU and memory that is provisioned for each node.')
+param sku string
+
+@description('Required. Disk storage size for the node group in GB.')
+param storage int
+
+@description('Optional. IP addresses to allow access to the cluster from.')
+param networkAcls networkAclsType?
+
+@description('Optional. Key vault reference and secret settings for the module\'s secrets export.')
+param secretsExportConfiguration secretsExportConfigurationType?
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints privateEndpointType
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  'Key Vault Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '00482a5a-887f-4fb3-b363-3b7fe8e74483'
+  )
+  'Key Vault Certificates Officer': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'a4417e6f-fecd-4de8-b567-7b0420556985'
+  )
+  'Key Vault Certificate User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'db79e9a7-68ee-4b58-9aeb-b90e7c24fcba'
+  )
+  'Key Vault Contributor': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'f25e0fa2-a7c8-4377-a976-54943a77a395'
+  )
+  'Key Vault Crypto Officer': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '14b46e9e-c2b7-41b4-b07b-48a6ebf60603'
+  )
+  'Key Vault Crypto Service Encryption User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'e147488a-f6f5-4113-8e2d-b22465e65bf6'
+  )
+  'Key Vault Crypto User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '12338af0-0e69-4776-bea7-57ae8d297424'
+  )
+  'Key Vault Reader': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '21090545-7ca7-4776-b22c-e363652d74d2'
+  )
+  'Key Vault Secrets Officer': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'b86a8fe4-44ce-4948-aee5-eccb2c155cd7'
+  )
+  'Key Vault Secrets User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '4633458b-17de-408a-b874-0445c86b69e6'
+  )
```

Comment:
Please replace those with roles relevant to the mongo-cluster resource

Reply:
Just tried to find some via `Get-AzRoleDefinition`, but it does not appear there are any specific to this resource. If the case, just can just remove the KeyVault roles and leave the other 5 standard roles (Owner etc.) in